client/allocwatcher/alloc_watcher.go

```diff
@@ -190,11 +190,18 @@ func (p *localPrevAlloc) Wait(ctx context.Context) error {
 
 	defer p.prevListener.Close()
 
+	// Don't bother blocking for updates from the previous alloc if it has
+	// already terminated.
+	if p.prevStatus.Terminated() {
```

Comment:
This is a defensive/fast-path to avoid even blocking on the alloc broadcaster if the previous alloc has terminated.